src/app.py

```diff
@@ -39,6 +39,41 @@
 app = FastAPI()
 
 
+class User:
+    def __init__(self, token: Optional[str]) -> None:
+        self.__token = token
+
+    @property
+    def is_anonymous(self) -> bool:
+        return self.__token is None
+
+    @property
+    def name(self) -> str:
+        if self.is_anonymous:
+            return "anonymous"
+        # This feature is still WIP. TODO - token parsing and validating
+        return "username"
+
+    @property
+    def roles(self) -> List[str]:
+        if self.is_anonymous:
+            # Temporarily enable everything to everyone
+            return ["loggedin", "admin"]
+        # This feature is still WIP. TODO - user roles and configuration
+        return ["loggedin", "admin"]
+
+
+async def current_user(authorization: Optional[str] = Header(None)) -> User:
+    if not authorization:
+        return User(None)
+
+    bearer, token = authorization.split()
```

Comment:
Not sure about handling this. Fails here should (probably?) lead to a 400 error, not 500